distributed/recreate_exceptions.py

```diff
@@ -124,6 +131,13 @@ def recreate_error_locally(self, future):
         Nothing; the function runs and should raise an exception, allowing
         the debugger to run.
         """
-        func, args, kwargs = sync(
-                self.client.loop, self._recreate_error_locally, future)
-        func(*args, **kwargs)
+        out = sync(self.client.loop, self._recreate_error_locally, future)
+        if len(out) == 3 and isinstance(out[2], dict):
+            func, args, kwargsfunc(*args, **kwargs)
+        else:
+            exec_tuple(out)
```

Comment:
It would be nice to replicate exactly whatever logic the worker does here.  Otherwise I expect we'll run into confusing corner cases.